### PR TITLE
Ensure torch.tensor and Tensor.new_tensor copy numpy data.

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Comment by gchanan** [comment]
We should probably separate the legacy stuff from tensor_new into tensor_new_legacy since this file is getting fairly complex.

File: test/test_torch.py [hunk 1]
```diff
@@ -1288,6 +1288,14 @@ def test_tensor_factory(self):
         self.assertEqual(res1, expected)
         self.assertIs(torch.int, res1.dtype)
 
+        # test copy with numpy
+        if TEST_NUMPY:
+            a = np.array([5.])
```
Comment (fmassa) [inline test/test_torch.py]: Does this return a `float64` array? If that's the case, the old code-path would also return a copy of the array, right?
Maybe it's better to impose that this is a `float32` `np.array`?

Comment (fmassa) [reply test/test_torch.py]: Nope, this is right, because we set the default tensor type to be `double` in `common.py`. Sorry for the noise

**Review by colesbury — APPROVED** [review]
yeah, let's clean this up soon

**Pull request merged**